builtin/plugins/dposv3/dpos.go

```diff
@@ -633,17 +633,20 @@ func (c *DPOS) claimRewardsFromAllValidators2(ctx contract.Context, req *ClaimDe
 	var claimedFromValidators []*types.Address
 	var amounts []*types.BigUInt
 	for _, d := range delegations {
-		if d.Index != REWARD_DELEGATION_INDEX {
+		if d.Index != REWARD_DELEGATION_INDEX || loom.UnmarshalAddressPB(d.Delegator).Compare(delegator) != 0 {
 			continue
 		}
+
 		delegation, err := GetDelegation(ctx, d.Index, *d.Validator, *delegator.MarshalPB())
 		if err == contract.ErrNotFound {
+			ctx.Logger().Error("DPOS error", "error", err, "sender", ctx.Message().Sender, "req", req)
 			continue
 		} else if err != nil {
-			return nil, err
+			return nil, errors.Wrap(err, "failed to load delegation")
 		}
 
-		if delegation.State == UNBONDING && delegation.UpdateAmount.Value.Cmp(&delegation.Amount.Value) == 0 {
+		if (delegation.State == UNBONDING && delegation.UpdateAmount.Value.Cmp(&delegation.Amount.Value) == 0) ||
+			delegation.Amount.Value.Cmp(common.BigZero()) == 0 {
```

Comment:
Instead of `delegation.Amount.Value.Cmp(common.BigZero()) == 0` you can write `delegation.Amount.Value.Sign() == 0`, slightly more concise.